contracts/near/eth-client/src/lib.rs

```diff
@@ -583,29 +662,42 @@ impl EthClient {
     }
 
     #[cfg(feature = "bsc")]
-    fn bsc_get_epoch_header(&self) -> BlockHeader {
-        self.headers.get(&self.epoch_header).unwrap()
+    fn bsc_get_validator_set_from_block(header: &BlockHeader) -> (Vec<u8>, u64) {
+        let validators = header.extra_data[BSC_EXTRA_VANITY..(header.extra_data.len() - BSC_EXTRA_SEAL)].to_vec();
+        let validators_len = (validators.len() / BSC_VALIDATOR_BYTES_SIZE) as u64;
+        (validators, validators_len)
+    }
+
+    #[cfg(feature = "bsc")]
+    fn bsc_get_current_validators(&self, header: &BlockHeader) -> (Vec<u8>, u64) {
+        let epoch_block_number = header.number - (header.number % BSC_EPOCH_SIZE as u64);
+        let prev_epoch_hash = self.canonical_header_hashes.get(&(epoch_block_number - BSC_EPOCH_SIZE as u64)).unwrap();
+        let prev_epoch_block = self.headers.get(&prev_epoch_hash).unwrap();
+        let (prev_validators, prev_validators_len) = EthClient::bsc_get_validator_set_from_block(&prev_epoch_block);
+
+        let is_from_prev_epoch = header.number <= header.number - (header.number % BSC_EPOCH_SIZE as u64) + (prev_validators_len) / 2;
```

Comment:
This could be simplified:

```suggestion
        let is_from_prev_epoch = header.number <= epoch_block_number + (prev_validators_len) / 2;
```